client/signup/controller.js

```diff
@@ -318,11 +308,13 @@ export default {
 		const signupDependencies = getSignupDependencyStore( getState() );
 
 		const siteIdOrSlug =
-			signupDependencies?.siteSlug ||
 			query?.siteSlug ||
-			signupDependencies?.siteId ||
-			query?.siteId;
+			query?.siteId ||
+			signupDependencies?.siteSlug ||
+			signupDependencies?.siteId;
```

Comment:
The order here was a bit weird, if a user puts a siteid or slug in the URL we should respect and use that - always.

Reply:
I think your instinct here to prefer the query param over the redux store makes sense. But I'd just want to check that this doesn't break our "choose domain" -> "checkout" -> "hero flow" progression, which is why we added this code in the first place.